homeassistant/components/light/wemo.py

```diff
@@ -140,3 +148,89 @@ def turn_off(self, **kwargs):
     def update(self):
         """Synchronize state with bridge."""
         self.update_lights(no_throttle=True)
+
+
+class WemoDimmer(Light):
+    """Representation of a WeMo dimmer."""
+
+    def __init__(self, device):
+        """Initialize the WeMo dimmer."""
+        self.wemo = device
+        self._brightness = None
+        self._state = None
+        # look up model name once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+
+        wemo = get_component('wemo')
+        wemo.SUBSCRIPTION_REGISTRY.register(self.wemo)
```

Comment:
Component resources should preferably be stored in the dict `hass.data`. We should consider if a refactor is appropriate in this PR.

Reply:
The majority of my code for adding the dimmer was actually written by others, just did some copy pasting from the switch/wemo.py to get it working. I'm definitely new to python, and coding in general, so I am not so sure I am the one you would want to try to get this cleaned up, because it would have to be cleaned up for the switch component as well, right?

Reply:
Yeah, ok. Then follow the rest of my comments, but leave the global attribute access.